kani-compiler/src/codegen_cprover_gotoc/codegen/assert.rs

```diff
@@ -18,6 +18,8 @@ pub enum PropertyClass {
     ExactDiv,
     ExpectFail,
     FiniteCheck,
+    /// Checks added by Kani compiler to detect UB or unstable behavior.
+    KaniCheck,
```

Comment:
Maybe rename to `UnstableCheck`? `KaniCheck` is not very descriptive.

Reply:
I was thinking we could use this class for all the checks we add and leave the assertion class only for user assertions. What do you think?

Reply:
Sounds good to me, but I think we'll want fine-grained control for some Kani-generated checks in the future. This is okay for now.

Reply:
We might, but I think it's too soon to break things down. I do think we should separate checks that we insert from user assertions though